### ISSUE TITLE
HoloLens 2 Emulator not launching, "Object Reference not set to an instance of an object" error

### ISSUE THREAD

**Issue by njwilsonISU** [MicrosoftDocs/mixed-reality#1283]
I have been trying to troubleshoot my issues for days, and tried everything listed above or anywhere else in this documentation(and elsewhere). I can install the emulator, sdk, and everything else fine, but when I attempt to deploy to the emulator it throws this error every time. Any suggestions?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0051f2fd-9bcd-7eb8-48f0-b8b170d42a49
* Version Independent ID: 2c1a0a2b-6a3f-1a1d-caf1-4a06caf5411b
* Content: [Using the HoloLens Emulator - Mixed Reality](https://docs.microsoft.com/en-us/windows/mixed-reality/using-the-hololens-emulator)
* Content Source: [mixed-reality-docs/using-the-hololens-emulator.md](https://github.com/MicrosoftDocs/mixed-reality/blob/master/mixed-reality-docs/using-the-hololens-emulator.md)
* Product: **mixed-reality**
* Technology: **mr-dev**
* GitHub Login: @pbarnettms
* Microsoft Alias: **pbarnett**

**Comment by pbarnettms** [comment]
@njwilsonISU Have you followed the steps in the Troubleshooting section of https://docs.microsoft.com/en-us/windows/mixed-reality/using-the-hololens-emulator?  Specifically, the second item under "If your installation completes successfully, but Visual Studio displays an error attempting to launch the HoloLens Emulator, please try the following" is likely the issue, and following the instructions there should get you past that.  If you still hit a problem after doing this, please let me know what version of Visual Studio you're using (e.g., 2017 15.9.12) and what version of Windows you're using (run 'ver' or 'winver').

**Comment by njwilsonISU** [comment]
Hi Paul, Thanks for the suggestion

I am certain that I have tried following every tutorial or troubleshooting guide listed under the mixed reality docs. After updating every application or package involved in the Unity workflow, the last thing I tried was to uninstall and reinstall everything listed on the "Install the tools" page (IN ORDER) and I can now deploy to the HoloLens but not the emulator. 

Here is my software summary:
Windows Pro 1809 (17763.503)
Unity 2018.4.2.f1
VS 2017 v15.9.13 
Windows 10 SDK (10.0.18362.0) 1903
HoloLens 2 Emulator build 10.0.18362.1019

**Comment by pbarnettms** [comment]
@njwilsonISU Does the HoloLens 2 Emulator start if you launch it from its Start menu shortcut?  (You can find this in the "Windows Kits" folder on your Start menu.  It will appear as "HoloLens 2 Emulator (10.0.18362.1019)".  If it does not start, do you never see the emulator window, or does it show an error?

Do you have the emulator installed to %ProgramFiles(x86)%\Windows Kits\10\?  Can you please send the full output from running this exact command in PowerShell?  (Please do not modify the command in any way, including the file path)

`(Get-Item -path "${env:ProgramFiles(x86)}\Windows Kits\10\Microsoft XDE\10.0.18362.0\XDE.exe").VersionInfo.ProductVersion`

**Comment by njwilsonISU** [comment]
@pbarnettms
'''PS C:\WINDOWS\system32> (Get-Item -path "${env:ProgramFiles(x86)}\Windows Kits\10\Microsoft XDE\10.0.18362.0\XDE.exe").VersionInfo.ProductVersion
Get-Item : Cannot find path 'C:\Program Files (x86)\Windows Kits\10\Microsoft XDE\10.0.18362.0\XDE.exe' because it
does not exist.
At line:1 char:2
+ (Get-Item -path "${env:ProgramFiles(x86)}\Windows Kits\10\Microsoft X ...
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Program File...18362.0\XDE.exe:String) [Get-Item], ItemNotFoundExcep
   tion
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetItemCommand'''

The emulator app when launched from the start menu does not open, it shows a window for a split second, then it closes.

**Comment by njwilsonISU** [comment]
```cmd
> (Get-Item -path "${env:ProgramFiles(x86)}\Windows Kits\10\Microsoft XDE\10.0.18362.0\XDE.exe").VersionInfo.ProductVersion
Get-Item : Cannot find path 'C:\Program Files (x86)\Windows Kits\10\Microsoft XDE\10.0.18362.0\XDE.exe' because it
does not exist.
At line:1 char:2
+ (Get-Item -path "${env:ProgramFiles(x86)}\Windows Kits\10\Microsoft X ...
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Program File...18362.0\XDE.exe:String) [Get-Item], ItemNotFoundExcep
   tion
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetItemCommand

```

When launching the app from the start menu, a window appears for a split second then closes.

**Comment by pbarnettms** [comment]
@njwilsonISU This indicates that the emulator is not installed to the correct location.  Your SDK might be elsewhere as well.  Visual Studio requires the emulator app (XDE.exe and its dependencies) to be in %ProgramFiles(x86)%\Windows Kits\10\Microsoft XDE\10.0.18362.0.  Can you please also send along the output of the following PowerShell command?

`Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows Kits\Installed Roots\" -Name "KitsRoot10"`

If the KitsRoot10 value is anything other than your x86 Program Files folder (e.g., "C:\Program Files (x86)\Windows Kits\10\"), we'll need to have you fix the value and reinstall the SDK and emulator.

**Comment by njwilsonISU** [comment]
@pbarnettms 
This is the output from powershell after running that command 
'''
KitsRoot10   : C:\Program Files (x86)\Windows Kits\10\
'''
So it seems this is correct but the emulator was just installed in the incorrect place, correct?
What is the solution to this problem?

**Comment by njwilsonISU** [comment]
@pbarnettms 

Every time I try to install the emulator with the installer given on this page or the archive page, it installs the package in C:\Program Files\ instead of C:\Program Files (x86)\
When I try to manually select the install path as C:\Program Files (x86)\Windows Kits\10\ an error occurs in the installation. I have uninstalled all the emulator versions and I am prepared to uninstall all associated programs and the sdk version and do a clean start, if that is what you recommend.

**Comment by pbarnettms** [comment]
@njwilsonISU What was the error that was displayed when you tried to install to Program Files (x86)?

Do you get the same value if you run the following?

`Get-ItemProperty -Path "HKLM:\SOFTWARE\WOW6432Node\Microsoft\Windows Kits\Installed Roots\" -Name "KitsRoot10"
`

The fact that it's installing to the wrong folder is the reason for the errors you're getting trying to start the emulator.  We'll need to address this.  The HoloLens Emulator will by default install to the same location as your Windows SDK, so your Windows SDK is likely incorrect as well.  I would recommend doing the following:

1. Uninstall all Windows SDKs and HoloLens Emulators, along with any other Microsoft SDKs you might have installed.  (Given the incorrect installation folder, you will likely encounter some problems with other Windows SDK components, though basic functionality should work).
2. Ensure that the KitsRoot10 value at both HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows Kits\Installed Roots\ and HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows Kits\Installed Roots\ point to your 32-bit Program Files folder (C:\Program Files (x86)... instead of C:\Program Files).
3. Launch the setup for the HoloLens 2 Emulator.  (It's ok to not have any other SDKs installed when you do this).
4. Before proceeding with the install, check the path to which it's going to install (the first path displayed, not the custom path option).  If it's anything other that C:\Program Files (x86)..., do not proceed with the installation.  Instead, please look in your Temp folder (C:\Users\<your user name>\AppData\Local\Temp\hololenssdk).  There should be at least one log file in there.  The one I'm most interested in is HoloLens_2_Emulator__Windows_10.0.18362.1019_<timestamp>_000_package_MicrosoftHoloLensEmulator_x64_en_us.log.  If you can make that available either by uploading it to online storage and providing a link, or by copying-and-pasting its contents into a reply, I should be able to help you fix the issue.

Thank you!

**Comment by njwilsonISU** [comment]
@pbarnettms 
ok so after uninstalling the emulator and all sdk's, I ran the prompts again...

```
PS W:\> Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows Kits\Installed Roots\" -Name "KitsRoot10"
>>
>>
KitsRoot10   : C:\Program Files\Windows Kits\10\
```

**Comment by pbarnettms** [comment]
@njwilsonISU Let's fix that value.  Please run the following from an Administrator command prompt:

`reg add "HKLM\Software\Microsoft\Windows Kits\Installed Roots" /v KitsRoot10 /t REG_SZ /d "C:\Program Files (x86)\Windows Kits\10\"`

(If prompted, answer 'Y' to overwrite the existing value).

Please then run the original PowerShell command again and confirm that it reports C:\Program Files (x86)...  Once it does, please continue with the emulator installation.  (You can also then install any other SDKs, though please run the PowerShell command before each one and let me know what SDK you just installed if you see the value change back to C:\Program Files\....)

**Comment by njwilsonISU** [comment]
@pbarnettms
I ran the command and verified that the change was made. then while attempting to install the 1903 SDK, an error occurred, with the following info
"The folder path '10"' contains an invalid character"

**Comment by pbarnettms** [comment]
@njwilsonISU It looks like the 'reg' command is treating the trailing double quote strangely.  Please try this:

reg add "HKLM\Software\Microsoft\Windows Kits\Installed Roots" /v KitsRoot10 /t REG_SZ /d "C:\Program Files (x86)\Windows Kits\10\

(Yes, the last double quote is missing - that's intentional, and should resolve the issue)

Once you've done this, please run the SDK setup again.

**Comment by njwilsonISU** [comment]
@pbarnettms 
I followed your steps and the errors mentioned above were corrected. 

BUT now when I try to run the emulator from the start menu or from the VS project, the emulator will start but a error box will appear and says

> [XDE:002] The emulator is unable to verify that the virtual machine is running:
> 
> Could not load the file or assembly 'Diskstream, Version=10.0.0.0, Culture=neurtal, PublicKeyToken=31bf3856ad364e35' or one of its dependencies.Strong name validation failed. (Exception from HRESULT: 0x8013141A)

I can start another comment thread if you would like

**Comment by pbarnettms** [comment]
@njwilsonISU We're hoping to post a new build soon that will resolve this issue, but we have a workaround you can use:

1. Open an Administrator command prompt.
2. Switch to the folder that contains sn.exe, which is part of the .NET Framework SDK. For example, this can be found in C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.7.2 Tools\x64
3. Run "sn -Vr Diskstream,31BF3856AD364E35"

After doing this, the emulator should now launch properly.

**Comment by pbarnettms** [comment]
Closing this issue as the workaround for 10.0.18362.1019 has been provided.  We're also releasing 10.0.18362.1021 this week which will address this issue and retroactively fix 10.0.18362.1019 if you have it installed.
